### PR TITLE
Implement custom ad notification field trials

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/15624

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Confirm the default position of custom ad notifications on macOS are positioned top-right to the left of system notifications
- Confirm the default ad notification timeout for custom ad notifications on macOS is 120 seconds
- Confirm the default position of custom ad notifications on Windows are positioned top-right
- Confirm the default ad notification timeout for custom ad notifications on Windows is 120 seconds
- Confirm the default position of custom ad notifications on Linux are positioned top-right
- Confirm the default ad notification timeout for custom ad notifications on Linux is 120 seconds
- Confirm the default ad notification timeout for custom ad notifications on Android is 30 seconds
- Confirm the default ad notification fade in/out animation duration for custom ad notifications is 200 milliseconds
- Confirm the default ad notification fade in/out animation duration for custom ad notifications is 200 milliseconds

Variations seed.json can be tested later as each change will need to go through the privacy team

### CODE REVIEW

File: components/brave_ads/browser/features.cc [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+const base::Feature kAdNotifications{"AdNotifications",
+                                     base::FEATURE_ENABLED_BY_DEFAULT};
+
+namespace {
+
+// Set to true to show custom ad notifications or false to show system
+// notifications
+const char kFieldTrialParameterShouldShowCustomAdNotifications[] =
+    "should_show_custom_notifications";
+const int kDefaultShouldShowCustomAdNotifications = true;
```
Comment (yachtcaptain23) [inline components/brave_ads/browser/features.cc:23]: Should this be boolean?
```suggestion
const bool kDefaultShouldShowCustomAdNotifications = true;
```

File: components/brave_ads/browser/features.cc [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+const char kFieldTrialParameterAdNotificationNormalizedDisplayCoordinateX[] =
+    "ad_notification_normalized_display_coordinate_x";
+#if defined(OS_WIN)
+const double kDefaultAdNotificationNormalizedDisplayCoordinateX = 1.0;
+#elif defined(OS_MAC)
+const double kDefaultAdNotificationNormalizedDisplayCoordinateX = 1.0;
+#elif defined(OS_LINUX)
+const double kDefaultAdNotificationNormalizedDisplayCoordinateX = 1.0;
+#elif defined(OS_ANDROID)
+const double kDefaultAdNotificationNormalizedDisplayCoordinateX = 0.5;
```
Comment (yachtcaptain23) [inline components/brave_ads/browser/features.cc:51]: Is this parameter used on Android? If not, would you put a comment mentioning that this parameter isn't used.

File: components/brave_ads/browser/features.cc (hunk 1 above)
Comment (tmancey) [reply components/brave_ads/browser/features.cc]: Great spot :-)

Comment (tmancey) [reply components/brave_ads/browser/features.cc]: Fixed

File: components/brave_ads/browser/features.cc (hunk 2 above)
Comment (tmancey) [reply components/brave_ads/browser/features.cc]: Refactored to remove unused code on Android

**Review by yachtcaptain23 — APPROVED** [review]

File: common/brave_channel_info.cc [hunk 3]
```diff
@@ -23,13 +23,4 @@ std::string GetChannelName() {
 #endif  // !OFFICIAL_BUILD
 }
 
-bool IsNightlyOrDeveloperBuild() {
```
Comment (moritzhaller) [inline common/brave_channel_info.cc:26]: should this be part of this PR?

Comment (tmancey) [reply common/brave_channel_info.cc:26]: This is related to adding variations as field trials replace the need for custom ad notifications to be pinned to nightly.

**Comment by tmancey** [comment]
post-init failed due to audit deps

**Pull request merged**